src/sas/qtgui/Perspectives/Fitting/FittingPerspective.py

```diff
@@ -263,9 +269,27 @@ def addFit(self, data, is_batch=False, tab_index=None):
         self.addTab(tab, icon, tab_name)
         # Show the new tab
         self.setCurrentWidget(tab)
+        # If configured, plot the current data
+        self.maybePlotOnLoad(tab)
+
         # Notify listeners
         self.tabsModifiedSignal.emit()
 
+    def maybePlotOnLoad(self,tab):
+        if tab.data_is_loaded and config.FITTING_PLOT_ON_SEND_DATA:
+            #First, create model and residuals inside this data.
+            def tidyup(self):
+                logger.info(msg='calculation complete, running tidy-up function')
+                #time.sleep(2)
+                #tab.onPlot()
+                #QtWidgets.QApplication.processEvents()
+                tab.showPlot()
+                #QtWidgets.QApplication.processEvents()
+                #time.sleep(2)
```

Comment:
Playing with `processEvents` means there were issues with plot lifetimes? What did you observe?